### PR TITLE
[WIP] Make use of Argu 3 features

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
This is a work in progress trying to make use of the new features found in Argu 3, most importantly subcommand support. Here's what running `paket.exe --help` looks like at the moment:
```
$ ./paket.exe --help
Paket version 3.4.0.0
Help was requested:
USAGE: paket.exe [add <subcommands>] [clear-cache <subcommands>] [config <subcommands>] [convert-from-nuget <subcommands>]
                 [find-refs <subcommands>] [init <subcommands>] [auto-restore <subcommands>] [install <subcommands>]
                 [outdated <subcommands>] [remove <subcommands>] [restore <subcommands>] [simplify <subcommands>]
                 [update <subcommands>] [find-packages <subcommands>] [find-package-versions <subcommands>]
                 [show-installed-packages <subcommands>] [show-groups <subcommands>] [pack <subcommands>]
                 [push <subcommands>] [generate-include-scripts <subcommands>] [--help]

SUBCOMMANDS:

        add <options>: Adds a new package to your paket.dependencies file.
        clear-cache <options>: Clears the NuGet and git cache folders.
        config <options>: Allows to store global configuration values like NuGet credentials.
        convert-from-nuget <options>: Converts from using NuGet to Paket.
        find-refs <options>: Finds all project files that have the given NuGet packages installed.
        init <options>: Creates an empty paket.dependencies file in the working directory.
        auto-restore <options>: Enables or disables automatic Package Restore in Visual Studio during the build process.
        install <options>: Download the dependencies specified by the paket.dependencies or paket.lock file into the `packages/` directory and update projects.
        outdated <options>: Lists all dependencies that have newer versions available.
        remove <options>: Removes a package from your paket.dependencies file and all paket.references files.
        restore <options>: Download the dependencies specified by the paket.lock file into the `packages/` directory.
        simplify <options>: Simplifies your paket.dependencies file by removing transitive dependencies.
        update <options>: Update one or all dependencies to their latest version and update projects.
        find-packages <options>: Allows to search for packages.
        find-package-versions <options>: Allows to search for package versions.
        show-installed-packages <options>: Shows all installed top-level packages.
        show-groups <options>: Shows all groups.
        pack <options>: Packs all paket.template files within this repository.
        push <options>: Pushes the given `.nupkg` file.
        generate-include-scripts <options>: Generate include scripts for installed packages.

        Use 'paket.exe <subcommand> --help' for additional information.

OPTIONS:

        --help: display this list of options.
```

### CODE REVIEW

File: docs/content/paket-push.md [hunk 1]
```diff
@@ -3,9 +3,9 @@
 Pushes the given `.nupkg` file.
 
     [lang=msh]
-    paket push fsi.exe url <string> file <string> [apikey <string>] [endpoint <string>] [--help]
+    paket push url <string> file <string> [apikey <string>] [endpoint <string>] [--help]
```
Comment (forki) [inline docs/content/paket-push.md]: Oups

Comment (eiriktsarpalis) [inline docs/content/paket-push.md]: What happened there? I didn't change this manually.

Comment (forki) [inline docs/content/paket-push.md]: yeah it looks like you fixed a bug ;-)

**Comment by eiriktsarpalis** [comment]
@forki I think I'm more or less done with my work here. CLI related integration tests seem to be passing. Here's what the usage text looks like now:
```
$ ./bin/paket.exe --help
Paket version 3.4.0.0
Help was requested:
USAGE: paket [--help] [--verbose] [--log-file <path>] [--silent] <subcommand> [<options>]

SUBCOMMANDS:

    add <options>         Adds a new package to your paket.dependencies file.
    clear-cache <options> Clears the NuGet and git cache folders.
    config <options>      Allows to store global configuration values like NuGet credentials.
    convert-from-nuget <options>
                          Converts from using NuGet to Paket.
    find-refs <options>   Finds all project files that have the given NuGet packages installed.
    init <options>        Creates an empty paket.dependencies file in the working directory.
    auto-restore <options>
                          Enables or disables automatic Package Restore in Visual Studio during the build process.
    install <options>     Download the dependencies specified by the paket.dependencies or paket.lock file into the `packages/` directory and update projects.
    outdated <options>    Lists all dependencies that have newer versions available.
    remove <options>      Removes a package from your paket.dependencies file and all paket.references files.
    restore <options>     Download the dependencies specified by the paket.lock file into the `packages/` directory.
    simplify <options>    Simplifies your paket.dependencies file by removing transitive dependencies.
    update <options>      Update one or all dependencies to their latest version and update projects.
    find-packages <options>
                          Allows to search for packages.
    find-package-versions <options>
                          Allows to search for package versions.
    show-installed-packages <options>
                          Shows all installed top-level packages.
    show-groups <options> Shows all groups.
    pack <options>        Packs all paket.template files within this repository.
    push <options>        Pushes the given `.nupkg` file.
    generate-include-scripts <options>
                          Generate include scripts for installed packages.

        Use 'paket <subcommand> --help' for additional information.

OPTIONS:

    --verbose, -v         Enable verbose console output for the paket process.
    --log-file <path>     Specify a log file for the paket process.
    --silent, -s          Suppress console output for the paket process.
    --help                display this list of options.
```
and in a subcommand:
```
$ ./bin/paket.exe pack --help
Paket version 3.4.0.0
Help was requested:
USAGE: paket pack [--help] output <path> [buildconfig <config name>] [buildplatform <target>] [version <version>]
                  [templatefile <path>] [exclude <templateId>] [specific-version <templateId> <version>]
                  [releaseNotes <text>] [lock-dependencies] [minimum-from-lock-file] [symbols] [include-referenced-projects]
                  [project-url <url>]

OPTIONS:

    output <path>         Output directory to put .nupkg files.
    buildconfig <config name>
                          Optionally specify build configuration that should be packaged (defaults to Release).
    buildplatform <target>
                          Optionally specify build platform that should be packaged (if not provided or empty, checks all known platform targets).
    version <version>     Specify version of the package.
    templatefile <path>   Allows to specify a single template file.
    exclude <templateId>  Exclude template file by id.
    specific-version <templateId> <version>
                          Specifies a version number for template with given id.
    releaseNotes <text>   Specify relase notes for the package.
    lock-dependencies     Get the version requirements from paket.lock instead of paket.dependencies.
    minimum-from-lock-file
                          Get the version requirements from paket.lock instead of paket.dependencies, and add them as a minimum version.  `lock-dependencies` will over-ride this option.
    symbols               Build symbol/source packages in addition to library/content packages.
    include-referenced-projects
                          Include symbol/source from referenced projects.
    project-url <url>     Url to the projects home page.
    --verbose, -v         Enable verbose console output for the paket process.
    --log-file <path>     Specify a log file for the paket process.
    --silent, -s          Suppress console output for the paket process.
    --help                display this list of options.
```

**Comment by forki** [comment]
thanks man, this is awesome!

**Pull request merged**